sdks/python/test-suites/portable/py2/build.gradle

```diff
@@ -78,6 +78,8 @@ task crossLanguagePythonJavaFlink {
     def options = [
         "--runner=PortableRunner",
         "--experiments=worker_threads=100",
+        // Need to add this to correctly translate bounded sources with an Impulse
+        "--experiments=beam_fn_api",
```

Comment:
The PR revealed some interesting issue with this flag, having to be added in case other experiments are present.

Reply:
Do we always have to add this flag when we have bounded sources? If so, can we set it automatically instead? https://github.com/apache/beam/blob/6454f724e6e1c266bc97a4ad8b99ad51a790924d/sdks/python/apache_beam/runners/portability/fn_api_runner.py#L345-L346

Reply:
It should be set where you linked above, at least when the runner has been set before the pipeline is created. Somewhere a Read without impulse still sneaks through. Have to debug further when I get a chance.

Reply:
We are using `PortableRunner` here, not `FnApiRunner`. It never appears to be set for `PortableRunner`, unless I am missing something.

Reply:
Ah, you linked a different place. There is one in `pipeline.py` as well: https://github.com/apache/beam/blob/4575a8007c1dd3af500c22042c7ceccb92ca8678/sdks/python/apache_beam/pipeline.py#L159

This will lead to the read being wrapped in Impulse + adapter: https://github.com/apache/beam/blob/4575a8007c1dd3af500c22042c7ceccb92ca8678/sdks/python/apache_beam/io/iobase.py#L866

Need to find the place where we still generate a raw bounded Read.

Reply:
Ok, it was not the flag. It was simply the fact that `PipelineTrimmer` would remove the subtransforms for bounded reads (Impulse + ParDo adapter) if Read was declared as a native transform. This would cause Reads with an environment to be root transforms.